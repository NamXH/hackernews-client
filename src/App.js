import React, { Component } from 'react';
import './App.css';

const BASE_QUERY = 'https://hn.algolia.com/api/v1/';
const SEARCH_QUERY = 'search?query=';

class App extends Component {

  constructor() {
    super();

    this.state = {
      hits: [],
      page: null,
      hitsPerPage: 0,
      query: 'redux',
      isLoading: true,
    };

    this.fetchSearchTopstories = this.fetchSearchTopstories.bind(this);
    this.onSearchChange = this.onSearchChange.bind(this);
    this.onSearchSubmit = this.onSearchSubmit.bind(this);
  }

  setSearchTopstories({ hits, page, hitsPerPage }) {
    this.setState({ hits, page, hitsPerPage, isLoading: false })
  }

  fetchSearchTopstories(query) {
    this.setState({ isLoading: true });
    fetch(`${BASE_QUERY}${SEARCH_QUERY}${query}`)
      .then(response => response.json())
      .then(result => this.setSearchTopstories(result));
  }

  onSearchChange(event) {
    this.setState({ query: event.target.value });
  }

  onSearchSubmit(event) {
    this.fetchSearchTopstories(this.state.query);
    event.preventDefault();
  }

  componentDidMount() {
    const { query } = this.state;
    this.fetchSearchTopstories(query);
  }

  render() {
    const { hits, query, isLoading } = this.state;

    return (
      <div>
        <form onSubmit={this.onSearchSubmit}>
          <input type="text" value={query} onChange={this.onSearchChange} />
          <button type="submit">Search</button>
        </form>
        <div>
          <span>Title</span>
          <span>Author</span>
          <span>Comment Count</span>
          <span>Points</span>
        </div>
        { isLoading ?
            <div>Loading ...</div> :
            hits.map((item, key) =>
              <div key={key}>
                <span><a href={item.url}>{item.title}</a></span>
                <span>{item.author}</span>
                <span>{item.num_comments}</span>
                <span>{item.points}</span>
              </div>
            )
        }
      </div>
    );
  }
}

export default App;
