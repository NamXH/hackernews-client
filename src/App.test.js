import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';
import { InputConfirm, MoreButton, Sort, HitsTable } from './App'

describe('App', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('InputConfirm', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<InputConfirm>Confirm Me</InputConfirm>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <InputConfirm>Confirm Me</InputConfirm>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Sort', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Sort>Sort Me</Sort>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Sort>Sort Me</Sort>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('MoreButton', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MoreButton>Give Me More</MoreButton>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <MoreButton>Give Me More</MoreButton>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('HitsTable', () => {

  const props = {
    results: {
      someResultKey: {
        hits: [
          { title: 'a', author: 'b', num_comments: 1, points: 2 }
        ],
      }
    },
    resultKey: 'someResultKey',
    sortKey: 'NONE',
  };

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HitsTable { ...props } />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <HitsTable { ...props } />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});