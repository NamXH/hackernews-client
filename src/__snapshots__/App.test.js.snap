exports[`App snapshots 1`] = `
<div
  className="page">
  <div
    className="interactions">
    <form
      onSubmit={[Function]}>
      <input
        onChange={[Function]}
        type="text"
        value="redux" />
      <button
        type="submit">
        Search
      </button>
    </form>
  </div>
  <div>
    <div
      className="table">
      <div
        className="table-header">
        <span
          style={
            Object {
              "width": "40%",
            }
          }>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button">
            Title
          </button>
        </span>
        <span
          style={
            Object {
              "width": "30%",
            }
          }>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button">
            Author
          </button>
        </span>
        <span
          style={
            Object {
              "width": "15%",
            }
          }>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button">
            Comments
          </button>
        </span>
        <span
          style={
            Object {
              "width": "15%",
            }
          }>
          <button
            className="button-inline"
            onClick={[Function]}
            type="button">
            Points
          </button>
        </span>
      </div>
      <div />
    </div>
  </div>
  <div
    className="interactions">
    <div>
      Loading ...
    </div>
  </div>
</div>
`;

exports[`HitsTable snapshots 1`] = `
<div
  className="table">
  <div
    className="table-header">
    <span
      style={
        Object {
          "width": "40%",
        }
      }>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button">
        Title
      </button>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button">
        Author
      </button>
    </span>
    <span
      style={
        Object {
          "width": "15%",
        }
      }>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button">
        Comments
      </button>
    </span>
    <span
      style={
        Object {
          "width": "15%",
        }
      }>
      <button
        className="button-inline"
        onClick={[Function]}
        type="button">
        Points
      </button>
    </span>
  </div>
  <div>
    <div
      className="table-row">
      <span
        style={
          Object {
            "width": "40%",
          }
        }>
        <a
          href={undefined}>
          a
        </a>
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }>
        b
      </span>
      <span
        style={
          Object {
            "width": "15%",
          }
        }>
        1
      </span>
      <span
        style={
          Object {
            "width": "15%",
          }
        }>
        2
      </span>
    </div>
  </div>
</div>
`;

exports[`InputConfirm snapshots 1`] = `
<form
  onSubmit={undefined}>
  <input
    onChange={undefined}
    type="text"
    value={undefined} />
  <button
    type="submit">
    Confirm Me
  </button>
</form>
`;

exports[`MoreButton snapshots 1`] = `
<button
  onClick={[Function]}
  type="button">
  Give Me More
</button>
`;

exports[`Sort snapshots 1`] = `
<button
  className="button-inline button-active"
  onClick={[Function]}
  type="button">
  Sort Me
</button>
`;
